src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java

```diff
@@ -33,23 +35,25 @@
 import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
 
 public final class BigQueryProject {
   private static final Logger logger = LoggerFactory.getLogger(BigQueryProject.class);
   private static final ConcurrentHashMap<String, BigQueryProject> PROJECT_CACHE =
       new ConcurrentHashMap<>();
   private final String projectId;
   private final BigQuery bigQuery;
-  private final int TIMEOUT_SECONDS = 40;
 
   private BigQueryProject(String projectId) {
     logger.info("Retrieving Bigquery project for project id: {}", projectId);
     this.projectId = projectId;
     HttpTransportOptions transportOptions = StorageOptions.getDefaultHttpTransportOptions();
+    ApplicationContext appCtx = ApplicationContextUtils.getApplicationContext();
+    GcsConfiguration gcsConfiguration = appCtx.getBean(GcsConfiguration.class);
     transportOptions =
         transportOptions.toBuilder()
-            .setConnectTimeout(TIMEOUT_SECONDS * 1000)
-            .setReadTimeout(TIMEOUT_SECONDS * 1000)
+            .setConnectTimeout(gcsConfiguration.getConnectTimeoutSeconds() * 1000)
```

Comment:
I think for this kind of thing we typically would write it as:
```suggestion
            .setConnectTimeout(TimeUnit.SECONDS.toMillis(gcsConfiguration.getConnectTimeoutSeconds()))
```

Reply:
Oh nice, that is better.